File: palindrome-number/palindrome-number.py
class Solution(object):
    def isPalindrome(self, x):
        """
        :type x: int
        :rtype: bool
        """
        if x < 0 or (x != 0 and x % 10 == 0):
            return False
        y = 0
        while x > y:
            y = y * 10 + x % 10
            x /= 10
        return x == y or x == y/10

File: palindrome-number/palindrome-number.js
/**
 * @param {number} x
 * @return {boolean}
 */
var isPalindrome = function(x) {
    if (x<0 || (x!=0 && x%10==0)) {
        return false;
    }
    var y = 0;
    while (x>y) {
        y = y*10 + x%10;
        x = parseInt(x/10);
    }
    return (x==y || x==parseInt(y/10));
};